mithril-common/src/digesters/cardano_immutable_digester.rs

```diff
@@ -49,99 +84,125 @@ impl ImmutableDigester for CardanoImmutableDigester {
+                    "error" => ?error
+                );
             }
-            Some(_) => {
-                let cached_values = match self.cache_provider.as_ref() {
-                    None => BTreeMap::from_iter(immutables.into_iter().map(|i| (i, None))),
-                    Some(cache_provider) => match cache_provider.get(immutables.clone()).await {
-                        Ok(values) => values,
-                        Err(error) => {
-                            warn!(
-                                self.logger, "Error while getting cached immutable files digests";
-                                "error" => ?error
-                            );
-                            BTreeMap::from_iter(immutables.into_iter().map(|i| (i, None)))
-                        }
-                    },
-                };
-
-                // digest is done in a separate thread because it is blocking the whole task
-                let logger = self.logger.clone();
-                let thread_cardano_network = self.cardano_network.clone();
-                let thread_beacon = beacon.clone();
-                let (hash, new_cache_entries) =
-                    tokio::task::spawn_blocking(move || -> CacheComputationResult {
-                        compute_hash(
-                            logger,
-                            thread_cardano_network,
-                            &thread_beacon,
-                            cached_values,
-                        )
-                    })
-                    .await
-                    .map_err(|e| ImmutableDigesterError::DigestComputationError(e.into()))??;
-                let digest = hex::encode(hash);
-
-                debug!(self.logger, "Computed digest: {digest:?}");
-
-                if let Some(cache_provider) = self.cache_provider.as_ref() {
-                    if let Err(error) = cache_provider.store(new_cache_entries).await {
-                        warn!(
-                            self.logger, "Error while storing new immutable files digests to cache";
-                            "error" => ?error
-                        );
-                    }
-                }
+        }
+
+        Ok(digest)
+    }
 
-                Ok(digest)
+    async fn compute_merkle_tree(
+        &self,
+        dirpath: &Path,
+        beacon: &CardanoDbBeacon,
+    ) -> Result<MKTree<MKTreeStoreInMemory>, ImmutableDigesterError> {
+        let immutables_to_process =
+            list_immutable_files_to_process(dirpath, beacon.immutable_file_number)?;
+        info!(self.logger, ">> compute_merkle_tree"; "beacon" => #?beacon, "nb_of_immutables" => immutables_to_process.len());
+        let computed_immutables_digests = self.process_immutables(immutables_to_process).await?;
+
+        let leaves: Vec<_> = computed_immutables_digests
+            .digests
+            .into_iter()
+            .map(|digest| MKTreeNode::new(digest.into()))
+            .collect();
```

Comment:
This conversion could probably be avoided here and left to the constructor of the Merkle tree which takes whatever is convertible into a MKTreeNode: `fn new<T: Into<MKTreeNode> + Clone>(leaves: &[T]) -> StdResult<Self>`.